crates/containerd-shim-wasm/Cargo.toml

```diff
@@ -28,6 +28,7 @@ clone3 = "0.2"
 libc = { workspace = true }
 caps = "0.5"
 proc-mounts = "0.3"
+libcontainer = { workspace = true }
```

Comment:
I think it might be bit of an issue that the `containerd-shim-wasm` crate needs to depend on `libcontainer` for error handling. Doesn't that imply that if an (external) shim uses some other error type, runwasi would also need to depend on the library that provides it to be consistent? I'm not really sure what would be the right solution here, and this PR isn't the problem, I'm just wondering if this could be done better.

Reply:
I would be interested in moving `youki` out of the WasmEdge implementation into a lower level of `runwasi` so that all shims can benefit from it. In that case having a dependency on `libcontainer` would be less of a concern.
I will open an Issue to discuss it further.

Reply:
Ok, it's a bit more complex thank I thought.

Adding this variant to the `Error` enum means that all the shims (not only those using `youki`)  inherit youki's runtime dependencies. This is basically `libdbus` (and transitively, `libsystemd`).

If we plan to move all shims to `youki`, then this is a reasonable cost to pay.
Otherwise I could place `youki`'s errors in `Error::Any`. It's less descriptive, but would do the trick without adding the extra dependencies.

Reply:
Just for this PR, is it possible to make containerd-shim-wasm not depend on youki?

Reply:
Yes, I think that's sensible. I'll use `Error::Any` for now.